From 85ae581146a9791fd45006b273c19f822f1ccf4f Mon Sep 17 00:00:00 2001
From: Sebastian Lackner <sebastian@fds-team.de>
Date: Sat, 1 Nov 2014 22:51:34 +0100
Subject: riched20: Silence repeated FIXMEs triggered by Adobe Reader.

Adobe Reader calls these functions very often while scrolling through a document.
---
 dlls/riched20/richole.c | 46 ++++++++++++++++++++++++++++++++++++----------
 1 file changed, 36 insertions(+), 10 deletions(-)

diff --git a/dlls/riched20/richole.c b/dlls/riched20/richole.c
index b9213f6..6676d9a 100644
--- a/dlls/riched20/richole.c
+++ b/dlls/riched20/richole.c
@@ -294,6 +294,14 @@ static HRESULT WINAPI IRichEditOleImpl_inner_fnQueryInterface(IUnknown *iface, R
         IUnknown_AddRef((IUnknown *)*ppvObj);
         return S_OK;
     }
+
+    if (IsEqualGUID(riid, &IID_ITextServices))
+    {
+        static int once;
+        if (!once++) FIXME("%p: unhandled interface IID_ITextServices\n", This);
+        return E_NOINTERFACE;
+    }
+
     FIXME("%p: unhandled interface %s\n", This, debugstr_guid(riid));
  
     return E_NOINTERFACE;   
@@ -1927,6 +1935,7 @@ static HRESULT WINAPI TextFont_SetBackColor(ITextFont *iface, LONG value)
 static HRESULT WINAPI TextFont_GetBold(ITextFont *iface, LONG *value)
 {
     ITextFontImpl *This = impl_from_ITextFont(iface);
+    static int once;
     TRACE("(%p)->(%p)\n", This, value);
 
     if (This->range)
@@ -1940,7 +1949,7 @@ static HRESULT WINAPI TextFont_GetBold(ITextFont *iface, LONG *value)
     if (!font_get_reole(This))
         return CO_E_RELEASED;
 
-    FIXME("not implemented\n");
+    if (!once++) FIXME("not implemented\n");
     return E_NOTIMPL;
 }
 
@@ -1980,7 +1989,9 @@ static HRESULT WINAPI TextFont_SetEmboss(ITextFont *iface, LONG value)
 static HRESULT WINAPI TextFont_GetForeColor(ITextFont *iface, LONG *value)
 {
     ITextFontImpl *This = impl_from_ITextFont(iface);
-    FIXME("(%p)->(%p): stub\n", This, value);
+    static int once;
+
+    if (!once++) FIXME("(%p)->(%p): stub\n", This, value);
 
     if (!value)
         return E_INVALIDARG;
@@ -2052,6 +2063,7 @@ static HRESULT WINAPI TextFont_SetEngrave(ITextFont *iface, LONG value)
 static HRESULT WINAPI TextFont_GetItalic(ITextFont *iface, LONG *value)
 {
     ITextFontImpl *This = impl_from_ITextFont(iface);
+    static int once;
     TRACE("(%p)->(%p)\n", This, value);
 
     if (This->range)
@@ -2065,7 +2077,7 @@ static HRESULT WINAPI TextFont_GetItalic(ITextFont *iface, LONG *value)
     if (!font_get_reole(This))
         return CO_E_RELEASED;
 
-    FIXME("not implemented\n");
+    if (!once++) FIXME("not implemented\n");
     return E_NOTIMPL;
 }
 
@@ -2105,7 +2117,9 @@ static HRESULT WINAPI TextFont_SetKerning(ITextFont *iface, FLOAT value)
 static HRESULT WINAPI TextFont_GetLanguageID(ITextFont *iface, LONG *value)
 {
     ITextFontImpl *This = impl_from_ITextFont(iface);
-    FIXME("(%p)->(%p): stub\n", This, value);
+    static int once;
+
+    if (!once++) FIXME("(%p)->(%p): stub\n", This, value);
 
     if (!font_get_reole(This))
         return CO_E_RELEASED;
@@ -2245,7 +2259,9 @@ static HRESULT WINAPI TextFont_SetShadow(ITextFont *iface, LONG value)
 static HRESULT WINAPI TextFont_GetSize(ITextFont *iface, FLOAT *value)
 {
     ITextFontImpl *This = impl_from_ITextFont(iface);
-    FIXME("(%p)->(%p): stub\n", This, value);
+    static int once;
+
+    if (!once++) FIXME("(%p)->(%p): stub\n", This, value);
 
     if (!value)
         return E_INVALIDARG;
@@ -2317,7 +2333,9 @@ static HRESULT WINAPI TextFont_SetSpacing(ITextFont *iface, FLOAT value)
 static HRESULT WINAPI TextFont_GetStrikeThrough(ITextFont *iface, LONG *value)
 {
     ITextFontImpl *This = impl_from_ITextFont(iface);
-    FIXME("(%p)->(%p): stub\n", This, value);
+    static int once;
+
+    if (!once++) FIXME("(%p)->(%p): stub\n", This, value);
 
     if (!value)
         return E_INVALIDARG;
@@ -2345,7 +2363,9 @@ static HRESULT WINAPI TextFont_SetStrikeThrough(ITextFont *iface, LONG value)
 static HRESULT WINAPI TextFont_GetSubscript(ITextFont *iface, LONG *value)
 {
     ITextFontImpl *This = impl_from_ITextFont(iface);
-    FIXME("(%p)->(%p): stub\n", This, value);
+    static int once;
+
+    if (!once++) FIXME("(%p)->(%p): stub\n", This, value);
 
     if (!value)
         return E_INVALIDARG;
@@ -2401,7 +2421,9 @@ static HRESULT WINAPI TextFont_SetSuperscript(ITextFont *iface, LONG value)
 static HRESULT WINAPI TextFont_GetUnderline(ITextFont *iface, LONG *value)
 {
     ITextFontImpl *This = impl_from_ITextFont(iface);
-    FIXME("(%p)->(%p): stub\n", This, value);
+    static int once;
+
+    if (!once++) FIXME("(%p)->(%p): stub\n", This, value);
 
     if (!value)
         return E_INVALIDARG;
@@ -2429,7 +2451,9 @@ static HRESULT WINAPI TextFont_SetUnderline(ITextFont *iface, LONG value)
 static HRESULT WINAPI TextFont_GetWeight(ITextFont *iface, LONG *value)
 {
     ITextFontImpl *This = impl_from_ITextFont(iface);
-    FIXME("(%p)->(%p): stub\n", This, value);
+    static int once;
+
+    if (!once++) FIXME("(%p)->(%p): stub\n", This, value);
 
     if (!font_get_reole(This))
         return CO_E_RELEASED;
@@ -2723,7 +2747,9 @@ static HRESULT WINAPI TextPara_SetStyle(ITextPara *iface, LONG value)
 static HRESULT WINAPI TextPara_GetAlignment(ITextPara *iface, LONG *value)
 {
     ITextParaImpl *This = impl_from_ITextPara(iface);
-    FIXME("(%p)->(%p)\n", This, value);
+    static int once;
+
+    if (!once++) FIXME("(%p)->(%p)\n", This, value);
 
     if (!para_get_reole(This))
         return CO_E_RELEASED;
-- 
2.4.0

